atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPool.java

```diff
@@ -518,7 +518,7 @@ private void refreshTokenRanges() {
                 if (isRetriableWithBackoffException(e)) {
                     log.warn("Retrying with backoff a query intended for host {}.", hostPool.getHost(), e);
                     try {
-                        Thread.sleep(numTries * 1000);
+                        Thread.sleep(numTries * ThreadLocalRandom.current().nextInt(1000));
```

Comment:
Couldn't this result in rather inconsistent backoffs? (e.g. sequences like 500ms, 200ms, 2970ms, 4ms)

Have you considered something still random, but guaranteed to increase, e.g. `numTries * 1000 + (random from 0..999)`?